lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -921,6 +1090,36 @@ func (c *ContainerProxy) createNewAttachClientWithTimeouts(connectTimeout, respo
 	return plClient, transport
 }
 
+func (c *ContainerProxy) createGzipTarClient(connectTimeout, responseTimeout, responseHeaderTimeout time.Duration) (*client.PortLayer, *httpclient.Transport) {
+
+	r := rc.New(c.portlayerAddr, "/", []string{"http"})
+	transport := &httpclient.Transport{
+		ConnectTimeout:        connectTimeout,
+		ResponseHeaderTimeout: responseHeaderTimeout,
+		RequestTimeout:        responseTimeout,
+	}
+
+	r.Transport = transport
+
+	plClient := client.New(r, nil)
+	r.Consumers["application/octet-stream"] = runtime.ByteStreamConsumer()
+	r.Producers["application/octet-stream"] = runtime.ByteStreamProducer()
+
+	r.Consumers["application/x-tgz"] = runtime.ConsumerFunc(func(rdr io.Reader, data interface{}) error {
+		bsc := runtime.ByteStreamConsumer()
+		gzReader, err := gzip.NewReader(tar.NewReader(rdr))
```

Comment:
Can this gzip reader be optional? From docker commit, I need both gzip format and tar format. If return from here with gzip format, I'll need two more times decompression to get blobsum, tar file size, and diffID, like what imagec did.
Feels like if I can get tar format file, it will be easier